_api-reference/security/configuration/get-configuration.md

```diff
@@ -0,0 +1,120 @@
+component: endpoints
+-->
+## Endpoints
+```json
+GET /_plugins/_security/api/securityconfig
+```
+<!-- spec_insert_end -->
+
+## Example request
+
+```bash
+GET /_plugins/_security/api/securityconfig
+```
+{% include copy-curl.html %}
+
+## Example response
+
+```json
+{
+  "config": {
+    "dynamic": {
+      "authc": {
+        "basic_internal_auth_domain": {
+          "http_enabled": true,
+          "transport_enabled": true,
+          "order": 0,
+          "http_authenticator": {
+            "challenge": true,
+            "type": "basic",
+            "config": {}
+          },
+          "authentication_backend": {
+            "type": "internal",
+            "config": {}
+          }
+        }
+      },
+      "authz": {
+        "roles_from_myldap": {
+          "http_enabled": true,
+          "transport_enabled": true,
+          "authorization_backend": {
+            "type": "ldap",
+            "config": {
+              "roles_search_filter": "(uniqueMember={0})",
+              "host": "ldap.example.com",
+              "port": 389
+            }
+          }
+        }
+      },
+      "multi_rolespan_enabled": true,
+      "hosts_resolver_mode": "ip-only",
+      "do_not_fail_on_forbidden": false
+    }
+  }
+}
+```
+
+## Response body fields
+
+The response body is a JSON object with the following fields:
+
+| Property | Data type | Description |
+| :--- | :--- | :--- |
+| `config` | Object | The root object containing the security configuration. |
+
+<details markdown="block">
+  <summary>
+    Response body fields: <code>config</code>
+  </summary>
+  {: .text-delta}
+
+`config` is a JSON object that contains the following fields:
+
+| Property | Data type | Description |
+| :--- | :--- | :--- |
+| `dynamic` | Object | The main configuration object containing all security configuration settings. Includes authentication domains (`authc`), authorization settings (`authz`), and various security behaviors. |
+
+</details>
+
+## Usage notes
+
+The Get Configuration API provides a way to inspect the current security configuration. When using the API, remember the following usage notes:
+
+- **Read-only operation**: This API only retrieves the configuration and does not modify it.
+
+- **Access control**: Access to this API should be restricted to administrators, as the configuration contains sensitive information about your security setup.
+
+## Security considerations
+
+The security configuration contains sensitive information about your authentication mechanisms, LDAP settings, and security policies. Consider the following security best practices:
+
+- Restrict access to this API to administrators only.
+- Be cautious about storing or logging the output from this API, as it may contain sensitive configuration details.
+- Use HTTPS/TLS when interacting with this API to prevent information disclosure.
+
+## Enabling this API
+
+By default, access to this API is restricted. To enable it, you need to:
```

Comment:
I think we should mention the general way access to this endpoint is granted.

In general, any users mapped to roles defined in the setting `plugins.security.restapi.roles_enabled: ["all_access", "security_rest_api_access"]` will have access to this API.